Makefile

```diff
@@ -1211,15 +1226,17 @@ tag-build:
 # -dev.cloud., then the tag-publish workflow is run with `cloud-only=true`. This can be
 # specified explicitly with `make tag-publish CLOUD_ONLY=<true|false>`.
 .PHONY: tag-publish
-tag-publish: CLOUD_ONLY = $(if $(HAS_CLOUD_SEMVER),true,false)
+tag-publish: CLOUD_ONLY = $(if $(IS_CLOUD_SEMVER),true,false)
+tag-publish: ENVIRONMENT = $(if $(IS_PROD_SEMVER),publish-prod,publish-stage)
```

Comment:
Since we're not ready for the tag-publish workflow to take `publish-stage` as an environment yet, perhaps a better approach to this is to just update the `gh` call to add `-f "environment=publish-prod"` (and `-f "environment=build-prod"` in `tag-build`) now and merge this PR after both the build and publish workflows work with stage?

Reply:
How about this for now, and revert later?
```suggestion
tag-publish: ENVIRONMENT = publish-prod
# tag-publish: ENVIRONMENT = $(if $(IS_PROD_SEMVER),publish-prod,publish-stage)
```

Stage publishing just won't work for now. I'll be working on publishing immediately after the other work lands (and backports are filed), so I don't think it's a huge deal.

Reply:
I'll put this PR on hold until the publishing can be done to stage. I think the two need to be consistent. Developers are building releases right now for which they want to access the build artifacts via publishing, so the two steps do need to be consistent.